extensions/binding/org.eclipse.smarthome.binding.dmx.test/src/test/java/org/eclipse/smarthome/binding/dmx/ChaserThingHandlerTest.java

```diff
@@ -112,4 +119,110 @@ public void initializationOfChaserThing() {
+        currentTime = dmxBridgeHandler.calcBuffer(currentTime, 1000);
+        waitForAssert(() -> assertThat(dmxBridgeHandler.getDmxChannelValue(100), is(equalTo(115))));
+    }
+
+    @Test
+    public void runningChaserWithResume() {
+        thingProperties.put(CONFIG_CHASER_STEPS, TEST_STEPS_REPEAT);
+        thingProperties.put(CONFIG_CHASER_RESUME_AFTER, true);
+        initialize();
+
+        dmxBridgeHandler.setDmxChannelValue(100, 193);
+
+        long currentTime = System.currentTimeMillis();
+        currentTime = dmxBridgeHandler.calcBuffer(currentTime, 0);
+
+        waitForAssert(() -> assertThat(dmxBridgeHandler.getDmxChannelValue(100), is(equalTo(193))));
+
+        chaserThingHandler.handleCommand(new ChannelUID(chaserThing.getUID(), CHANNEL_SWITCH), OnOffType.ON);
+
+        // step I
+        currentTime = dmxBridgeHandler.calcBuffer(currentTime, 1000);
+        waitForAssert(() -> assertThat(dmxBridgeHandler.getDmxChannelValue(100), is(equalTo(115))));
+        currentTime = dmxBridgeHandler.calcBuffer(currentTime, 1000);
+        waitForAssert(() -> assertThat(dmxBridgeHandler.getDmxChannelValue(100), is(equalTo(115))));
+        // step II
+        currentTime = dmxBridgeHandler.calcBuffer(currentTime, 1000);
+        waitForAssert(() -> assertThat(dmxBridgeHandler.getDmxChannelValue(100), is(equalTo(210))));
+        currentTime = dmxBridgeHandler.calcBuffer(currentTime, 1000);
+        waitForAssert(() -> assertThat(dmxBridgeHandler.getDmxChannelValue(100), is(equalTo(210))));
+        // resume old state
+        currentTime = dmxBridgeHandler.calcBuffer(currentTime, 1000);
+        waitForAssert(() -> assertThat(dmxBridgeHandler.getDmxChannelValue(100), is(equalTo(193))));
+    }
+
+    private void initialize() {
+        chaserThing = ThingBuilder.create(THING_TYPE_CHASER, "testchaser").withLabel("Chaser Thing")
+                .withBridge(bridge.getUID()).withConfiguration(new Configuration(thingProperties)).build();
+
+        managedThingProvider.add(bridge);
+        dmxBridgeHandler = (TestBridgeHandler) waitForAssert(() -> {
+            final ThingHandler thingHandler = bridge.getHandler();
+            assertThat(thingHandler, notNullValue());
+            return thingHandler;
+        });
+
+        managedThingProvider.add(chaserThing);
+        chaserThingHandler = (ChaserThingHandler) waitForAssert(() -> {
+            final ThingHandler thingHandler = chaserThing.getHandler();
+            assertThat(thingHandler, notNullValue());
+            return thingHandler;
+        });
+
+        final ManagedItemChannelLinkProvider itemChannelLinkProvider = waitForAssert(() -> {
+            final ManagedItemChannelLinkProvider tmp = getService(ManagedItemChannelLinkProvider.class);
+            assertThat(tmp, is(notNullValue()));
+            return tmp;
+        });
+
+        switchItem = new DimmerItem(TEST_SWITCHITEM_NAME);
```

Comment:
At first I stumbled across this line because you assign a `Dimmeritem` to a variable called "switch....", but then I was thinking why you need the items at all for your tests?

Background of my question is that I see you testing basically our whole framework and we have a lot of JUnit tests "randomly failing" (in the past also some of your DMX tests) because of timing issues and how the framework behaves in tests.

Wouldn't it be better to just test your logic by calling your methods without the framework? and for the parts where you need the frame work do not create real registries or items but instead use mocks?

Reply:
DimmerItem is obviously wrong :-). Can you point me to an example for the mocks? I dind't find out how to do that.

Reply:
Found it. Will Update the checks.

Reply:
@J-N-K Regarding the mocks... You want to test your `ThingHandler` and we just found out that the framework does not allow easy mocking because we are using `ServiceTracker`s so you cannot easily mock away the registries for example. 

Since you found something, I am looking forward to what you have achieved, but otherwise I would suggest that you keep it as it is for now we will get back to you once we got rid of the `ServiceTracker`s and have provided a decent example. Then you could create a follow up PR to clean up your tests. But this is out of scope for this PR.

Reply:
see [here](https://github.com/J-N-K/smarthome/blob/testing/extensions/binding/org.eclipse.smarthome.binding.dmx.test/src/test/java/org/eclipse/smarthome/binding/dmx/DimmerThingHandlerTest.java).

Reply:
Can you push it as a new commit onto this branch please?

Reply:
Sure. Please see the last commit here.